ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1026,7 +1026,7 @@ private struct ZipState<Left, Right> {
 	var completed: (left: Bool, right: Bool) = (false, false)
 
 	var isFinished: Bool {
-		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty)
+		return (completed.left && values.left.isEmpty) || (completed.right && values.right.isEmpty) || (completed.left && completed.right)
```

Comment:
> If completed.left && completed.right and the previous conditions aren't true, that means that there are more values that should have been sent out.

Then `isFinished` would be false anyway, and the logic looking up for values works independent of this `isFinished` flag.